ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -629,11 +629,27 @@ static KsqlRestApplication buildApplication(
 
     final KsqlSecurityExtension securityExtension = loadSecurityExtension(ksqlConfig);
 
+    final Vertx vertx = Vertx.vertx(
+        new VertxOptions()
+            .setMaxWorkerExecuteTimeUnit(TimeUnit.MILLISECONDS)
+            .setMaxWorkerExecuteTime(Long.MAX_VALUE)
+            .setMetricsOptions(setUpHttpMetrics(ksqlConfig)));
+    vertx.exceptionHandler(t -> log.error("Unhandled exception in Vert.x", t));
+
+    final KsqlClient sharedClient = new KsqlClient(
+        toClientProps(ksqlConfig.originals()),
+        Optional.empty(),
+        new LocalProperties(ImmutableMap.of()),
+        new HttpClientOptions().setMaxPoolSize(100),
+        vertx
+    );
+
     final KsqlSecurityContextProvider ksqlSecurityContextProvider =
         new DefaultKsqlSecurityContextProvider(
             securityExtension,
             RestServiceContextFactory::create,
-            RestServiceContextFactory::create, ksqlConfig, schemaRegistryClientFactory);
+            RestServiceContextFactory::create, ksqlConfig, schemaRegistryClientFactory,
+            sharedClient);
```

Comment:
Does this mean that the shared client will be used for all internode communication? Also heartbeat and lag reporting?